core/src/test/java/tech/tablesaw/table/TableTransposeTest.java

```diff
@@ -0,0 +1,167 @@
+    DoubleColumn value2 = DoubleColumn.create("value2", new double[]{ 2.0, 2.1, 2.2});
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
+        new Object[][] {
+            {"label", "row1", "row2", "row3"},
+            {"value1", 1.0, 1.1, 1.2},
+            {"value2", 2.0, 2.1, 2.2},
+        },
+        result);
+    assertEquals(testTable.name(), result.name());
+    assertEquals(testTable.print(), result.transpose().print());
+  }
+
+  @Test
+  void transposeWithMissingData() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
+    DoubleColumn value = DoubleColumn.create("value1", new double[] { 1.0, DoubleColumnType.missingValueIndicator(), 1.2 });
+    DoubleColumn value2 = DoubleColumn.create("value2", new double[] { 2.0, 2.1, 2.2 });
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
+        new Object[][] {
+            {"label", "row1", "row2", "row3"},
+            {"value1", 1.0, Double.NaN, 1.2},
+            {"value2", 2.0, 2.1, 2.2},
+        },
+        result);
+
+    assertEquals(testTable.print(), result.transpose().print());
+  }
+
+  @Test
+  void transposeFloats() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
+    FloatColumn value = FloatColumn.create("value1", new float[] { 1.0f, 1.1f, 1.2f });
+    FloatColumn value2 = FloatColumn.create("value2", new float[] { 2.0f, 2.1f, 2.2f});
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+    assertEquals(testTable.print(), result.transpose().print());
+  }
+
+  @Test
+  void transposeIntegers() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
+    IntColumn value = IntColumn.create("value1", new int[]{ 1, 2, 3 });
+    IntColumn value2 = IntColumn.create("value2", new int[]{ 4, 5, 6 });
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
+        new Object[][] {
+            {"label", "row1", "row2", "row3"},
+            {"value1", 1, 2, 3},
+            {"value2", 4, 5, 6}
+        },
+        result);
+
+    assertEquals(testTable.print(), result.transpose().print());
+  }
+
+  @Test
+  void transposeLongs() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
+    LongColumn value = LongColumn.create("value1", new long[]{ 1, 2, 3 });
+    LongColumn value2 = LongColumn.create("value2", new long[] { 4, 5, 6});
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
+        new Object[][] {
+            {"label", "row1", "row2", "row3"},
```

Comment:
I would not expect the column names to end up in the contents

Reply:
This is the correct implementation of transpose in my view. Everything is kept, even column names, but flipped.  The first row (the headings) becomes the first column. 
This is consistent with Python pandas and Excel
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.transpose.html
If someone has a use case where they want to get rid of that first column, they can drop it in a subsequent step

Reply:
I don't think it really is consistent with Pandas. Pandas doesn't move the labels into the dataframe contents. Instead it has an additional concept of row names and swaps the column and row names.

I think we could also go the other way and say if someone has a usecase where they want the labels to be in the table contents they can add it in a subsequent step. It'd be slightly more code, but not too much more and I think it'd be much more expected. It looks like @ryancerf was also confused by this https://github.com/jtablesaw/tablesaw/pull/717#issuecomment-552737591.